source/common/network/address_impl.h

```diff
@@ -165,6 +183,17 @@ class Ipv4Instance : public InstanceBase {
                         const SocketInterface* sock_interface = nullptr,
                         absl::optional<std::string> network_namespace = absl::nullopt);
 
+  /**
+   * Construct from an existing unix IPv4 socket address (IP v4 address and port) for metadata
+   * purposes. Store the status code in passed in parameter instead of throwing.
+   * @param skip_validation if true, skip operating system address family validation. This should
+   * only be true when creating addresses from protocol headers (e.g., PROXY protocol) that will
+   * not be used for actual socket operations.
+   */
+  explicit Ipv4Instance(absl::Status& error, const sockaddr_in* address,
+                        const SocketInterface* sock_interface,
+                        absl::optional<std::string> network_namespace, bool skip_validation);
```

Comment:
Does this need to be a new constructor? Can it just be another optional parameter on the existing constructor?